src/Catalyst.Core.Modules.KeySigner/KeySigner.cs

```diff
@@ -55,18 +54,18 @@ public sealed class KeySigner : IKeySigner
         /// <inheritdoc/>
         ICryptoContext IKeySigner.CryptoContext => _cryptoContext;
 
-        public IPublicKey GetPublicKey(KeyRegistryTypes keyIdentifier)
-        {
-            var publicKey = (Ed25519PublicKeyParameters) _keyApi.GetPublicKeyAsync("self").GetAwaiter().GetResult();
-            var publicKeyBytes = publicKey.GetEncoded();
-            return _cryptoContext.GetPublicKeyFromBytes(publicKeyBytes);
-        }
-
         public IPrivateKey GetPrivateKey(KeyRegistryTypes keyIdentifier)
         {
-            var privateKey = (Ed25519PrivateKeyParameters) _keyApi.GetPrivateKeyAsync("self").GetAwaiter().GetResult();
-            var privateKeyBytes = privateKey.GetEncoded();
-            return _cryptoContext.GetPrivateKeyFromBytes(privateKeyBytes);
+            if (_keyRegistry.RegistryContainsKey(keyIdentifier))
+            {
+                return _keyRegistry.GetItemFromRegistry(keyIdentifier);
+            }
+
+            var privateKeyParameters = (Ed25519PrivateKeyParameters) _keyApi.GetPrivateKeyAsync(keyIdentifier.Name).GetAwaiter().GetResult();
+            var privateKeyBytes = privateKeyParameters.GetEncoded();
+            var privateKey = _cryptoContext.GetPrivateKeyFromBytes(privateKeyBytes);
+
+            return privateKey;
```

Comment:
@Xela101 Should the key registry be populated if it doesn't already contain the key here?